File: ProgrammingClass6.Angular/programmingclass6.angular.client/src/app/shared/models/product.ts
import { Manufacturer } from "./manufacturer";
import { ProductType } from "./productType";
import { UnitOfMeasure } from "./unitOfMeasure";

export interface Product {
  id?: number;
  name?: string;
  description?: string;
  quantity?: number;
  unitPrice?: number;

  manufacturer?: Manufacturer;
  productType?: ProductType;
  unitOfMeasure?: UnitOfMeasure;
}

File: ProgrammingClass6.Angular/ProgrammingClass6.Angular.Server/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace ProgrammingClass6.Angular.Server.Models
{
    public class Product
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        [StringLength(500)]
        public string Description { get; set; }

        public int Quantity { get; set; }

        public decimal UnitPrice { get; set; }

        public int? ManufacturerId { get; set; }

        public Manufacturer Manufacturer { get; set; }

        public int? ProductTypeId { get; set; }

        public ProductType ProductType { get; set; }

        public int? UnitOfMeasureId { get; set; }

        public UnitOfMeasure UnitOfMeasure { get; set; }
    }
}
